src/utils.zig

```diff
@@ -29,6 +32,33 @@ pub const CharacterType = enum {
         };
     }
 
+    pub fn fromUnicode(c: u21, allocator: std.mem.Allocator) CharacterType {
+        const cd = CaseData.init(allocator) catch @panic("Memory error");
+        defer cd.deinit();
+        const gcd = GenCatData.init(allocator) catch @panic("Memory error");
```

Comment:
Again, this feels like we should be able to do it with a stack allocated buffer instead of a `std.mem.Allocator` to avoid alloc overheads.